index.js

```diff
@@ -58,88 +58,108 @@ const runChecks = async (context, checkEvent) => {
-            await checkPullRequestLabelsModule.checkForIssueLabel(context);
+            callable.push(
+              checkPullRequestLabelsModule.checkForIssueLabel(context)
+            );
             break;
           case constants.datastoreLabelCheck:
-            await checkPullRequestLabelsModule.checkCriticalLabel(context);
+            callable.push(
+              checkPullRequestLabelsModule.checkCriticalLabel(context)
+            );
             break;
           case constants.forcePushCheck:
-            await checkBranchPushModule.handleForcePush(context);
+            callable.push(checkBranchPushModule.handleForcePush(context));
             break;
           case constants.prTemplateCheck:
-            await checkPullRequestTemplateModule.checkTemplate(context);
+            callable.push(
+              checkPullRequestTemplateModule.checkTemplate(context)
+            );
             break;
           case constants.pullRequestReviewCheck:
-            await checkPullRequestReviewModule.handlePullRequestReview(context);
+            callable.push(
+              checkPullRequestReviewModule.handlePullRequestReview(context)
+            );
             break;
           case constants.codeOwnerCheck:
-            await newCodeOwnerModule.checkForNewCodeowner(context);
+            callable.push(newCodeOwnerModule.checkForNewCodeowner(context));
             break;
           case constants.ciFailureCheck:
-            await ciCheckModule.handleFailure(context);
+            callable.push(ciCheckModule.handleFailure(context));
             break;
           case constants.updateWithDevelopCheck:
-            await checkMergeConflictsModule
-              .pingAllPullRequestsToMergeFromDevelop(
+            callable.push(
+              checkMergeConflictsModule.pingAllPullRequestsToMergeFromDevelop(
                 context
-              );
+              )
+            );
             break;
           case constants.periodicCheck:
-            await Promise.all([
+            callable.push(...[
               periodicCheckModule.ensureAllPullRequestsAreAssigned(context),
               periodicCheckModule.ensureAllIssuesHaveProjects(context),
-              staleBuildModule.checkAndTagPRsWithOldBuilds(context)
+              staleBuildModule.checkAndTagPRsWithOldBuilds(context),
             ]);
             break;
           case constants.respondToReviewCheck:
-            await checkPullRequestReviewModule.handleResponseToReview(context);
+            callable.push(
+              checkPullRequestReviewModule.handleResponseToReview(context)
+            );
             break;
           case constants.oldBuildLabelCheck:
-            await staleBuildModule.removeOldBuildLabel(context);
+            callable.push(staleBuildModule.removeOldBuildLabel(context));
             break;
         }
       }
+      // Wait for all checks to resolve or reject.
+      await Promise.allSettled(callable);

```

Comment:
This was needed because without this, any failure from a feature would prevent other features from running. This way, a feature's failure does not stop other features from running.